src/lib/OS/Process.cc

```diff
@@ -396,7 +396,7 @@ void Process::initContext(
   context_.TID = TID_;
   context_.pc = entryPoint_;
   context_.sp = stackPtr;
-  context_.progByteLen = getProcessImageSize();
+  context_.dataSectionEnd = getHeapStart();
```

Comment:
With the virtual memory setup, there is a padding between of `PAGE_SIZE` bytes between the maximum address of the `.data` section and heap. Do you think it can lead to problems in the future? Would `getHeapStart - PAGE_SIZE` work?

Reply:
Good spot, changed